source/service.go

```diff
@@ -260,7 +260,20 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 	}
 	sort.Strings(headlessDomains)
 	for _, headlessDomain := range headlessDomains {
-		targets := targetsByHeadlessDomain[headlessDomain]
+		allTargets := targetsByHeadlessDomain[headlessDomain]
+		targets := []string{}
+
+		deduppedTargets := map[string]bool{}
```

Comment:
instead of bool I would use struct{}, it seems it performs slightly better

Reply:
@njuettner i appreciate the swift review! updated.